doc/user/content/sql/show-types.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "SHOW TYPES"
+description: "`SHOW TYPES` returns a list of the data types in your Materialize instance."
+menu:
+  main:
+    parent: 'sql'
+---
+
+`SHOW TYPES` returns a list of the data types in your Materialize instance. By default, only user-created types are returned.
```

Comment:
I'm sure this isn't consistent everywhere, but this PR refers to user-created types and custom types; I'd stick with just calling them one thing. I slightly prefer `custom types` because it avoids the hyphen but I defer.